internal/clients/influxdb.go

```diff
@@ -55,6 +58,36 @@ func NewClient(ctx context.Context, kube client.Client, mg resource.Managed) (in
 	return influxdbv2.NewClient(pc.Spec.Endpoint, string(token)), nil
 }
 
+// NewClientWithResponses returns the bare client. Use this only if NewClient
+// does not meet your needs.
+func NewClientWithResponses(ctx context.Context, kube client.Client, mg resource.Managed) (*domain.ClientWithResponses, error) {
+	pc := &v1alpha1.ProviderConfig{}
+	if err := kube.Get(ctx, types.NamespacedName{Name: mg.GetProviderConfigReference().Name}, pc); err != nil {
+		return nil, errors.Wrap(err, errGetPC)
+	}
+
+	if err := resource.NewProviderConfigUsageTracker(kube, &v1alpha1.ProviderConfigUsage{}).Track(ctx, mg); err != nil {
+		return nil, errors.Wrap(err, errTrackPCUsage)
+	}
+
+	cd := pc.Spec.Credentials
+	token, err := resource.CommonCredentialExtractor(ctx, cd.Source, kube, cd.CommonCredentialSelectors)
+	if err != nil {
+		return nil, errors.Wrap(err, errGetCreds)
+	}
+	normServerURL := pc.Spec.Endpoint
```

Comment:
This line and afterwards is copied from `influxdbv2.NewClient`

Reply:
nit: maybe we should add this as a comment for future reference.